calcom/cal.com
# feat: Add syntax highlighting and autocomplete menu for workflow variables

## What does this PR do?


This adds syntax highlighting as well as notion-like autocomplete menu for workflow variables.
<img width="1512" alt="Screenshot 2024-06-02 at 15 57 12" src="https://github.com/calcom/cal.com/assets/70239736/4d7ed9a9-760e-4030-97ba-814915116022">


- Fixes https://github.com/calcom/cal.com/issues/15285
- Fixes [CAL-3847]


 Loom Video: https://www.loom.com/share/c4f920e667ea490f9bca9bc64d1868a4?sid=82d7f2bb-d47c-49e8-9617-6d88ce43a0fa

## Mandatory Tasks (DO NOT REMOVE)


## How should this be tested?



- Create a new workflow with an email action go to the text editor and press `{`, you should get a notion-like autocomplete menu with all the available variables.
- Try adding custom variables, by default anything wrapped within `{` `}` will get highlighted as a variable
